pkg/kn/core/root.go

```diff
@@ -71,11 +72,33 @@ Eventing: Manage event subscriptions and channels. Connect up event sources.`,
 	rootCmd.AddCommand(commands.NewCompletionCommand(p))
 	rootCmd.AddCommand(commands.NewVersionCommand(p))
 
+	// Deal with empty and unknown sub command groups
+	emptyAndUnknownSubcommands(rootCmd)
+
 	// For glog parse error.
 	flag.CommandLine.Parse([]string{})
 	return rootCmd
 }
 
+// emptyAndUnknownSubcommands adds a RunE to all commands that are groups to
+// deal with errors when called with empty or unknown sub command
+func emptyAndUnknownSubcommands(cmd *cobra.Command) {
+	for _, childCmd := range cmd.Commands() {
+		if childCmd.HasSubCommands() && childCmd.RunE == nil {
+			childCmd.RunE = func(aCmd *cobra.Command, args []string) error {
+				aCmd.Help()
+				if len(args) == 0 {
+					return errors.New(fmt.Sprintf("please provide a valid sub-command for \"kn %s\"", aCmd.Name()))
+				} else {
+					return errors.New(fmt.Sprintf("unknown command \"%s\" for \"kn %s\"", args[0], aCmd.Name()))
```

Comment:
*unkown sub-command

Reply:
Hmm, did not see the changes to the docs. I’ll try again tomorrow.

Yes, once we agree I will add the tests as we agree on the route for tests.

Reply:
OK addressed this in latest push. Resolving. Re-open if I did not. Thx.